frame/system/src/lib.rs

```diff
@@ -548,6 +586,11 @@ decl_module! {
 		}
 
 		/// Kill all storage items with a key that starts with the given prefix.
+		/// 
+		/// # <weight>
+		/// - `O(P)` where `P` amount of keys with prefix `prefix`
```

Comment:
I think this is not O(P). I think it is O(1). Basically we just remove the prefix node, and later pruning will do the rest. Would be good to confirm.

Reply:
AFAICT `clear_prefix` in the externalities calls
+ `clear_prefix` in the overlay which iterates over all prospective and commited changes and clears the ones with the prefix (which might be more expensive?)
+ `self.backend.for_keys_with_prefix` which iterates over the keys with that prefix to then do `self.overlay.set_storage`.
I guess that doesn't have to cash out into `P` database writes, but the runtime seems to at least be bounded by `O(P)`? Or do we just ignore the iteration done in the overlay here because it's dwarved by the db writes?

Also isn't the worst case still that the db needs to do that amount of deletes? After all there could be 100 entries with that prefix that get deleted in this block by this call.

Reply:
Oh, you mean that in the database, those 100 entries might be deleted by removing the prefix node. Seems to depend on the DB implementation then, though, right?

Reply:
not really I think. There is a trie layer above the DB, and this is what we care about.

You can find the correct answer here by running a test (make a mock runtime, with a benchmark, and I can show you how to check the DB logs), or ask the question in our public channel. But I do not think this is O(P) (but could be wrong). Just have someone confirm :)

Reply:
So, let's leave it as is?